statsmodels/regression/tests/test_lme.py

```diff
@@ -1094,6 +1096,62 @@ def test_random_effects_getters():
         assert(refc[g].shape == (p, p))
 
 
+def test_smw_solver():
+
+    np.random.seed(23)
+    p = 5
+    q = 4
+    r = 2
+    s = 2
+
+    A = np.random.normal(size=(p, q))
+    AtA = np.dot(A.T, A)
+
+    B = np.zeros((q, q))
+    B[0:r, 0:r] = np.random.normal(size=(r, r))
+    di = np.random.uniform(size=s)
+    B[r:q, r:q] = np.diag(1/di)
+    Qi = np.linalg.inv(B[0:r, 0:r])
+    s = 0.5
+
+    x = np.random.normal(size=p)
+    y2 = np.linalg.solve(s*np.eye(p, p) + np.dot(A, np.dot(B, A.T)), x)
+
+    f = _smw_solver(s, A, AtA, Qi, di)
+    y1 = f(x)
+    assert_allclose(y1, y2)
+
+    f = _smw_solver(s, sparse.csr_matrix(A), sparse.csr_matrix(AtA), Qi, di)
+    y1 = f(x)
+    assert_allclose(y1, y2)
+
+
+def test_smw_logdet():
+
+    np.random.seed(23)
+    p = 5
+    q = 4
+    r = 2
+    s = 2
+
+    A = np.random.normal(size=(p, q))
+    AtA = np.dot(A.T, A)
+
+    B = np.zeros((q, q))
+    c = np.random.normal(size=(r, r))
+    B[0:r, 0:r] = np.dot(c.T, c)
+    di = np.random.uniform(size=s)
+    B[r:q, r:q] = np.diag(1/di)
+    Qi = np.linalg.inv(B[0:r, 0:r])
+    s = 0.5
```

Comment:
@kshedden thanks for putting this together.  I'm interested in putting together something like hypothesis-style tests for this (recall I'm ultimately interested in debugging #4414).  To that end: is the construction above effectively the general case for constructing valid inputs for these two functions>\?

Reply:
Yes, I think this test is fairly general, except for the dimensions which are hard-coded.

Reply:
Thanks.  Any restrictions on valid (or reasonable) dimensions?

Reply:
No restrictions, it should work for a wide variety of dimensions.

Reply:
Substituting in `np.random.randint(0, 5)` for each of `p, q, r, s` it looks like we need the restriction `q = r + s`.  The dimension errors that pop up after that are harder to pin down; I'll list them here as I figure them out.

Reply:
It looks like the setup in test_smw_logdet is slightly different from test_smw_solver:

```
    c = np.random.normal(size=(r, r))
    B[0:r, 0:r] = np.dot(c.T, c)
```

vs

```
    B[0:r, 0:r] = np.random.normal(size=(r, r))
```

Is that intentional?  My naive prior is that the setups would be identical.

Reply:
Bounding `p >= 1` fixes all the failures in test_smw_logdet; but test_smw_solver fails (raises with mismatches dimensions) in ~40% of cases.